core/src/executor/evaluate/function.rs

```diff
@@ -645,6 +645,41 @@ pub fn generate_uuid<'a>() -> Evaluated<'a> {
     Evaluated::from(Value::Uuid(Uuid::new_v4().as_u128()))
 }
 
+pub fn greatest(_name: String, exprs: Vec<Evaluated<'_>>) -> Result<Evaluated<'_>> {
```

Comment:
to handle this function, it looks that we can do a bit different approach than other functions.

`Evaluated` provides `evaluate_cmp` method which can compare with other `Evaluated` value.
Then, rather than converting the evaluated into `Value`, we can simply use `Evaluated::evaluate_cmp` to compare each other.
`Evaluated::evaluate_cmp` returns `None` on non-comparable pair so we can still use `CannotCompareDifferentTypes` error on `None` return value.